comp/core/gui/guiimpl/gui.go

```diff
@@ -56,74 +79,111 @@ type Payload struct {
+func newGui(deps dependencies) (optional.Option[guicomp.Component], error) {
+
+	guiPort := deps.Config.GetString("GUI_port")
+
+	if guiPort == "-1" {
+		deps.Log.Infof("GUI server port -1 specified: not starting the GUI.")
+		return optional.NewNoneOption[guicomp.Component](), nil
+	}
+
+	g := gui{
+		port:      guiPort,
+		logger:    deps.Log,
+		config:    deps.Config,
+		flare:     deps.Flare,
+		status:    deps.Status,
+		collector: deps.Collector,
+		ac:        deps.Ac,
+	}
 
 	// Instantiate the gorilla/mux router
-	router := mux.NewRouter()
+	g.router = mux.NewRouter()
 
 	// Serve the only public file at the authentication endpoint
-	router.HandleFunc("/authenticate", generateAuthEndpoint)
+	g.router.HandleFunc("/authenticate", g.generateAuthEndpoint)
 
 	// Serve the (secured) index page on the default endpoint
-	router.Handle("/", authorizeAccess(http.HandlerFunc(generateIndex)))
+	g.router.Handle("/", g.authorizeAccess(http.HandlerFunc(generateIndex)))
 
 	// Mount our (secured) filesystem at the view/{path} route
-	router.PathPrefix("/view/").Handler(http.StripPrefix("/view/", authorizeAccess(http.HandlerFunc(serveAssets))))
+	g.router.PathPrefix("/view/").Handler(http.StripPrefix("/view/", g.authorizeAccess(http.HandlerFunc(serveAssets))))
 
 	// Set up handlers for the API
 	agentRouter := mux.NewRouter().PathPrefix("/agent").Subrouter().StrictSlash(true)
-	agentHandler(agentRouter, flare, statusComponent)
+	g.agentHandler(agentRouter)
 	checkRouter := mux.NewRouter().PathPrefix("/checks").Subrouter().StrictSlash(true)
-	checkHandler(checkRouter, collector, ac)
+	g.checkHandler(checkRouter)
 
 	// Add authorization middleware to all the API endpoints
-	router.PathPrefix("/agent").Handler(negroni.New(negroni.HandlerFunc(authorizePOST), negroni.Wrap(agentRouter)))
-	router.PathPrefix("/checks").Handler(negroni.New(negroni.HandlerFunc(authorizePOST), negroni.Wrap(checkRouter)))
+	g.router.PathPrefix("/agent").Handler(negroni.New(negroni.HandlerFunc(g.authorizePOST), negroni.Wrap(agentRouter)))
+	g.router.PathPrefix("/checks").Handler(negroni.New(negroni.HandlerFunc(g.authorizePOST), negroni.Wrap(checkRouter)))
 
-	// Listen & serve
-	listener, e := net.Listen("tcp", "127.0.0.1:"+port)
-	if e != nil {
-		return e
-	}
-	go http.Serve(listener, router) //nolint:errcheck
-	log.Infof("GUI server is listening at 127.0.0.1:" + port)
+	deps.Lc.Append(fx.Hook{
+		OnStart: g.start,
+		OnStop:  g.stop})
 
 	// Create a CSRF token (unique to each session)
-	e = createCSRFToken()
+	e := g.createCSRFToken()
 	if e != nil {
-		return e
+		return optional.NewNoneOption[guicomp.Component](), e
 	}
 
 	// Fetch the authentication token (persists across sessions)
-	authToken, e = security.FetchAuthToken(pkgconfig.Datadog)
+	g.authToken, e = security.FetchAuthToken(deps.Config)
 	if e != nil {
-		listener.Close()
-		listener = nil
+		return optional.NewNoneOption[guicomp.Component](), e
 	}
```

Comment:
Should we move this early return to the top of the `newGui` function? 

We are registering the lifecycle events and handlers, but if we hit this error cases we are returning an optional Gui component. 

I think is best practice to avoid doing extra work if an early return condition is found